pandas/tests/indexes/test_numeric.py

```diff
@@ -919,6 +919,10 @@ def test_constructor(self):
         res = Index([1, 2**63])
         tm.assert_index_equal(res, idx)
 
+        idx = Index([-1, 2**63], dtype=object)
```

Comment:
With regards to the specifying `uint64` with negative or `NaN`, the current behavior is this:
~~~python
>>> Index([-1], dtype='uint64')
UInt64Index([18446744073709551615], dtype='uint64')
>>>
>>> Index([np.nan], dtype='uint64')
Float64Index([nan], dtype='float64')
~~~
`uint64` does not raise on either case (which I imagine is where we are / should be going is because) we return `np.array([-1],..., dtype='uint64')`, and `numpy` is too kind that it casts `-1` to `uint64`.  For `np.nan`, when it detects floating data, it tries converting to an integer index before giving up and returning `Float64Index`.

I should point out though that `int64` has similar behavior for `NaN` (same `__new__`):
~~~
>>> Index([np.nan], dtype='int64')
Float64Index([nan], dtype='float64')
~~~

~~~